### PR TITLE
MSBuild terminal logger enamblement.

### PR DESCRIPTION
**Pull request by rokonec**
Fixes: part of https://github.com/dotnet/msbuild/issues/9063

Changes done:
-  prepare for easy enablement in dotnet 9
- support new msbuild logging configuration telemetry
- minor refactoring

### CODE REVIEW

**Review by rainersigwald — APPROVED** [review]
LGTM but needs SDK approval

File: src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildForwardingAppWithoutLogging.cs
Comment (rokonec) [reply src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildForwardingAppWithoutLogging.cs:20]: I see, but `DOTNET_CLI_USE_MSBUILD_TERMINAL_LOGGER` implies boolean, like use or dontuse, but this variable has null | true | false | on | off | auto states. 
Would `DOTNET_CLI_CONFIGURE_MSBUILD_TERMINAL_LOGGER` be better fitting? Not that it matter as mach as most people whom will need it will simple copy paste from docs page.

**Comment by baronfel** [comment]
Retargeted to the rc1 branch now that it's been created. Once this merges, we'll need an explicit backport to release/8.0.1xx as well (no auto-codeflow from RC1 to stable).

File: src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildForwardingAppWithoutLogging.cs
Comment (baronfel) [reply src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildForwardingAppWithoutLogging.cs:20]: `DOTNET_CLI_CONFIGURE_MSBUILD_TERMINAL_LOGGER` is acceptable to me

**Review by baronfel — APPROVED** [review]
Approving to unblock merge once the env var rename is made.

File: src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildForwardingAppWithoutLogging.cs
Comment (rokonec) [reply src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildForwardingAppWithoutLogging.cs]: done

**Pull request merged**
